src/libproc_macro/lib.rs

```diff
@@ -200,6 +201,72 @@ impl Span {
     pub fn call_site() -> Span {
         ::__internal::with_sess(|(_, mark)| Span(mark.expn_info().unwrap().call_site))
     }
+
+    /// The original source file into which this span points.
+    #[unstable(feature = "proc_macro", issue = "38356")]
+    pub fn file(&self) -> SourceFile {
```

Comment:
Wouldn't the source file be the file in which the other macro was defined then?
I still think this should be `Option<SourceFile>`, at least for `impl FromStr for TokenStream` (i.e. `DUMMY_SP`).